docker/docker.go

```diff
@@ -41,12 +41,20 @@ func (service *InfoService) GetInformationFromDockerEngine() (map[string]string,
 	info[agent.MemberTagKeyNodeName] = dockerInfo.Name
 
 	if dockerInfo.Swarm.NodeID == "" {
```

Comment:
To make this block clearer I would advise extracting its content into two unexported functions:

```go
	if dockerInfo.Swarm.NodeID == "" {
		getStandaloneInformation(info)
	} else {
		err := getSwarmInformation(info)
		if err != nil {
			return nil, err
		}
	}
```

The content of the `isLeaderNode` function could be integrated into the `getSwarmInformation` function.